test: test_oversize_body_returns_400 is flaky on CI (BrokenPipeError)

## Symptom

`tests/test_server.py::test_oversize_body_returns_400` fails intermittently on GitHub Actions with:

```
>       status, body = post(f"{base}/agent-auth/v1/validate", raw=payload)
E       BrokenPipeError: [Errno 32] Broken pipe
```

Observed in Test run [24667015232](https://github.com/aidanns/agent-auth/actions/runs/24667015232) on PR #139 at commit b2cae1f. Passes locally and passed on the underlying source-branch CI (`aidanns/token-routes-v1`, run 24666655259, same `server.py`). A rerun on the same commit cleared it, so the input is not bad — the server is sometimes closing the connection mid-send rather than reading the full body.

## Why it matters

Intermittent CI failures cause unnecessary reruns on unrelated PRs (this was observed on a documentation-only PR that touched no code). If the server's "reject oversize body" path is actually racing with the client's `sendall`, fixing the handler to drain the body cleanly before sending 400 would remove the flake.

## Suggested next steps

- Reproduce deterministically by tightening the test (e.g. loopback-only socket send with a specific timing pattern).
- If the server closes the socket without reading the full body, restructure the handler to read-then-reject so clients receive a 400 rather than a TCP reset.

## Related

- Surfaced by PR #139 (observability: adopt OTel semconv). The PR itself did not change the server or the test.

--- a/src/agent_auth/server.py
+++ b/src/agent_auth/server.py
@@ -58,6 +58,19 @@ class AgentAuthHandler(BaseHTTPRequestHandler):
         if length == 0:
             return {}
         if length > self.MAX_BODY_SIZE:
+            # Drain the announced body before sending 400. Closing the
+            # socket mid-upload surfaces a BrokenPipeError on the
+            # client's sendall() instead of a clean status response —
+            # the observed flake in #144. Read in fixed chunks rather
+            # than buffering the whole body so an oversize
+            # Content-Length header can't still exhaust memory; the
+            # bytes are discarded.
+            remaining = length
+            while remaining > 0:
+                chunk = self.rfile.read(min(remaining, 65536))
+                if not chunk:
+                    break
+                remaining -= len(chunk)
             return None
         body = self.rfile.read(length)
         try:

